### PR TITLE
Daily message limit per user

### LINKED ISSUE #2415: [FEAT]: Be able to set daily conversations per user instead of global setting
### What would you like to see?

Can we move this global setting in `/settings/system-preferences`

Screenshot 2024-10-03 at 11 05 21 AM

To become a 'per user' limit under `/settings/users`->edit ?

Screenshot 2024-10-03 at 11 06 04 AM

### PR DESCRIPTION
**Pull request by shatfield4**
### Pull Request Type

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] 📝 docs

### Relevant Issues

resolves #2415 

### What is in this change?

- Fix typos in username validation text
- Remove old functionality of restricting daily message limit for all users except admins
- Implement restricting daily message limit for users
- Remove unused endpoints
- Remove general settings page since no longer needed (only setting on this page was restricting daily message limit for all users)

### Developer Validations

- [x] I ran `yarn lint` from the root of the repo & committed changes
- [x] Relevant documentation has been updated
- [x] I have tested my code functionality
- [x] Docker build succeeds locally

### CODE REVIEW

File: server/models/user.js [hunk 1]
```diff
@@ -32,12 +33,17 @@ const User = {
       }
       return String(role);
     },
+    dailyMessageLimit: (dailyMessageLimit = null) => {
+      return dailyMessageLimit === null ? null : Number(dailyMessageLimit);
+    },
```
Comment (timothycarambat) [inline server/models/user.js:38]: Should be >= 1

**Review by timothycarambat — CHANGES_REQUESTED** [review]

**Comment by daaddd** [comment]
I tried to update but couldn't check out the changes, did I miss a step here?

```
sudo docker stop anythingllm-container

sudo docker rm anythingllm-container

sudo docker rmi mintplexlabs/anythingllm:latest

sudo docker pull mintplexlabs/anythingllm:latest

sudo docker run -d -p 3001:3001 --cap-add SYS_ADMIN \
-v /home/ec2-user/anythingllm:/app/server/storage \
-v /home/ec2-user/anythingllm/.env:/app/server/.env \
-e STORAGE_DIR="/app/server/storage" \
--name anythingllm-container mintplexlabs/anythingllm:latest
```

File: server/models/user.js (hunk 1 above)
Comment (shatfield4) [reply server/models/user.js:38]: For this, we use the `dailyMessageLimit` field to store if the option is enabled/disabled and the message limit. The way it currently works is if the option is off, it sets the field to `NULL` but when it is enabled we set it to the int value.

**Review by timothycarambat — APPROVED** [review]

**Comment by daaddd** [comment]
I see that this is now merged, I re-ran the commands above pulling from `latest and still the update didn't take place, do I need to use a specific version number?

```
sudo docker stop anythingllm-container

sudo docker rm anythingllm-container

sudo docker rmi mintplexlabs/anythingllm:latest

sudo docker system prune -af

sudo docker pull mintplexlabs/anythingllm:latest

sudo docker run -d -p 3001:3001 --cap-add SYS_ADMIN \
-v /home/ec2-user/anythingllm:/app/server/storage \
-v /home/ec2-user/anythingllm/.env:/app/server/.env \
-e STORAGE_DIR="/app/server/storage" \
--name anythingllm-container mintplexlabs/anythingllm:latest
```

**Comment by timothycarambat** [comment]
>  still the update didn't take place

How are you sure of this? It has been in latest for a month now and can be set per user when you are an `admin` role in the system

**Comment by daaddd** [comment]
> How are you sure of this?

Screenshot 2024-11-19 at 3 17 28 PM
fd8-a81daf5a2b0b">

I'm also not seeing `grok` as a valid provider in the LLM selection which seemed to be another one of the new features , is there any command I can run on my end to verify why the update didn't take place? Thank you

**Comment by timothycarambat** [comment]
running 
```
sudo docker pull mintplexlabs/anythingllm:latest
```

Should be all you need since you are running docker as sudo. Another way to check is 
```
docker inspect IMAGE_ID
```

This should show more metadata about the image to see if it truly pulled from repo or not.
https://hub.docker.com/layers/mintplexlabs/anythingllm/latest/images/sha256-cddb1498a87f7d16311fb9ad5dcbe868390d50ad8c80d6c771d0ff2c9e5ddbb3?context=repo  shows last push 2 hours ago, which make sense

**Comment by daaddd** [comment]
Thank you, this is the output I got:

```
[ec2-user@ip-172-31-25-138 ~]$ sudo docker inspect fb1d5a238fee
[
    {
        "Id": "fb1d5a238fee175b7cc9725c36db43acd665813e19c2ac494ed8447a82fccef0",
        "Created": "2024-11-19T19:27:06.09885533Z",
        "Path": "/bin/bash",
        "Args": [
            "/usr/local/bin/docker-entrypoint.sh"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 16981,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2024-11-19T19:27:07.433238831Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "healthy",
                "FailingStreak": 0,
                "Log": [
                    {
                        "Start": "2024-11-19T20:54:14.374913459Z",
                        "End": "2024-11-19T20:54:14.462565851Z",
                        "ExitCode": 0,
                        "Output": "Server is up\n"
                    },
                    {
                        "Start": "2024-11-19T20:55:14.465278852Z",
                        "End": "2024-11-19T20:55:14.537663671Z",
                        "ExitCode": 0,
                        "Output": "Server is up\n"
                    },
                    {
                        "Start": "2024-11-19T20:56:14.538190183Z",
                        "End": "2024-11-19T20:56:14.609665904Z",
                        "ExitCode": 0,
                        "Output": "Server is up\n"
                    },
                    {
                        "Start": "2024-11-19T20:57:14.612348464Z",
                        "End": "2024-11-19T20:57:14.693708327Z",
                        "ExitCode": 0,
                        "Output": "Server is up\n"
                    },
                    {
                        "Start": "2024-11-19T20:58:14.694378478Z",
                        "End": "2024-11-19T20:58:14.761510361Z",
                        "ExitCode": 0,
                        "Output": "Server is up\n"
                    }
                ]
            }
        },
        "Image": "sha256:bd7de7db66a5005f175564dc73310fb967bc92cf7c7568bbcc1424475287bccc",
        "ResolvConfPath": "/var/lib/docker/containers/fb1d5a238fee175b7cc9725c36db43acd665813e19c2ac494ed8447a82fccef0/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/fb1d5a238fee175b7cc9725c36db43acd665813e19c2ac494ed8447a82fccef0/hostname",
        "HostsPath": "/var/lib/docker/containers/fb1d5a238fee175b7cc9725c36db43acd665813e19c2ac494ed8447a82fccef0/hosts",
        "LogPath": "/var/lib/docker/containers/fb1d5a238fee175b7cc9725c36db43acd665813e19c2ac494ed8447a82fccef0/fb1d5a238fee175b7cc9725c36db43acd665813e19c2ac494ed8447a82fccef0-json.log",
        "Name": "/anythingllm-container",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": [
            "c0c739867447215467d58c3dba0b8258c8288801cc999cc6e2b9bac5f15eab22"
        ],
        "HostConfig": {
            "Binds": [
                "/home/ec2-user/anythingllm:/app/server/storage",
                "/home/ec2-user/anythingllm/.env:/app/server/.env"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {
                "3001/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "3001"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "no",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "ConsoleSize": [
                46,
                221
            ],
            "CapAdd": [
                "SYS_ADMIN"
            ],
            "CapDrop": null,
            "CgroupnsMode": "host",
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": [],
            "BlkioDeviceWriteBps": [],
            "BlkioDeviceReadIOps": [],
            "BlkioDeviceWriteIOps": [],
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": [
                {
                    "Name": "nofile",
                    "Hard": 65536,
                    "Soft": 32768
                }
            ],
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware",
                "/sys/devices/virtual/powercap"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/ed1e23866863223da72b698fe5997d2d5496d385d7adebb3f39e086af495d39c-init/diff:/var/lib/docker/overlay2/7c8498c0fdc67889c89be6856d786ca0f6b35ef92a1cb2b2532ac9cc81914297/diff:/var/lib/docker/overlay2/d5480cb17f003cbdb55c59537f366b2ad580f2db2eb8288203823383067b771a/diff:/var/lib/docker/overlay2/a9892a91e624b5108d2df6fbf2d74e6ca628365389ba50bea367036f6fae8830/diff:/var/lib/docker/overlay2/2ebafc61a08d5836fa29c507912e00d20096247ad63fbfafdbf2fa12d493a24a/diff:/var/lib/docker/overlay2/24f232faa08c13bded255c54a071c1a7ba6157e54e6fac5fc56f2ae3edd98723/diff:/var/lib/docker/overlay2/dd201a76bd4bd3bf9101a4d6fa351a8c31f25f6a9b22b4d038c435a0150dfdb5/diff:/var/lib/docker/overlay2/b360c8aada0ebe742bd0b02a0cf0aab3ad670216247b82a2b9984a99385fdad4/diff:/var/lib/docker/overlay2/9d9eeb00170246291c9e0793abfa97db42c32db36dae93c5dc9364ecbc230ebd/diff:/var/lib/docker/overlay2/d5baca95ce6a7165946d2309d93b60bc6d39fc3c06bd6e2b4bd54819ea217122/diff:/var/lib/docker/overlay2/0e9f0535a659cd18d45782cbb83d75e771306a63d232ce56449e0671cdba6da9/diff:/var/lib/docker/overlay2/dee9b643c7fd96739f802db2ae15bccbd125f37c472785f08168a606aeda6ee2/diff:/var/lib/docker/overlay2/4f07056310ccd8dd682a2b0f50e6ebb7ad260c184b980aa6134973c8165db6b0/diff:/var/lib/docker/overlay2/4f33b3fd53793d364a6b2a749483b6b38f81b2b52cd201d88dbf524cebb3e64b/diff:/var/lib/docker/overlay2/8783b7d0becaebb97b917b90854412a638c78b160ca582300f4855c9c3f72782/diff:/var/lib/docker/overlay2/beae67537efb57ca9284c9ade48a185c2e27e77f36a55799df5ceca6c2b766f6/diff:/var/lib/docker/overlay2/09bf9f294d6e7a0c0cd48b24dfd0ed375bde0cd5a5dd15eb06879403cf372f24/diff:/var/lib/docker/overlay2/d11be8a244ba80e031334396db9547a67b310ace923f0adeca13b376261a2f7c/diff:/var/lib/docker/overlay2/355b6fbcaa25353bce4a834a349ee66759d9b78b15b23801ac8268fef97e5148/diff:/var/lib/docker/overlay2/3b033d6bc7f9cc5529dc40f283add311b7682de878061c78b455ebd149c56c4b/diff:/var/lib/docker/overlay2/d08a520a5cf354f19284d808e29679c9f4a944bf747626bbf4a31c7e634ab761/diff:/var/lib/docker/overlay2/3e9b79f2f7e0d8efb56ee4d4def31d8d11142d531e16bcb353f1a391f27279df/diff:/var/lib/docker/overlay2/9a8d25e29de3e41787cb0ad0fb5c632af3da5f494eea743b055ef40af21dcf54/diff:/var/lib/docker/overlay2/59a80a0a81c7ee279c4f014f6d2ca7935b0ed536ab50a7dfde68fd50a91f3273/diff",
                "MergedDir": "/var/lib/docker/overlay2/ed1e23866863223da72b698fe5997d2d5496d385d7adebb3f39e086af495d39c/merged",
                "UpperDir": "/var/lib/docker/overlay2/ed1e23866863223da72b698fe5997d2d5496d385d7adebb3f39e086af495d39c/diff",
                "WorkDir": "/var/lib/docker/overlay2/ed1e23866863223da72b698fe5997d2d5496d385d7adebb3f39e086af495d39c/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/home/ec2-user/anythingllm",
                "Destination": "/app/server/storage",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/home/ec2-user/anythingllm/.env",
                "Destination": "/app/server/.env",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "fb1d5a238fee",
            "Domainname": "",
            "User": "anythingllm",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "3001/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "STORAGE_DIR=/app/server/storage",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "PUPPETEER_DOWNLOAD_BASE_URL=https://storage.googleapis.com/chrome-for-testing-public",
                "NODE_ENV=production",
                "ANYTHING_LLM_RUNTIME=docker"
            ],
            "Cmd": null,
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "/bin/bash /usr/local/bin/docker-healthcheck.sh || exit 1"
                ],
                "Interval": 60000000000,
                "Timeout": 10000000000,
                "StartPeriod": 60000000000
            },
            "Image": "mintplexlabs/anythingllm:1.2.4",
            "Volumes": null,
            "WorkingDir": "/app",
            "Entrypoint": [
                "/bin/bash",
                "/usr/local/bin/docker-entrypoint.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "org.opencontainers.image.created": "2024-10-25T00:03:20.778Z",
                "org.opencontainers.image.description": "The all-in-one Desktop & Docker AI application with built-in RAG, AI agents, and more.",
                "org.opencontainers.image.licenses": "MIT",
                "org.opencontainers.image.ref.name": "ubuntu",
                "org.opencontainers.image.revision": "72ba9f7f289047eac3bf1199a7e6892bf40bb3da",
                "org.opencontainers.image.source": "https://github.com/Mintplex-Labs/anything-llm",
                "org.opencontainers.image.title": "anything-llm",
                "org.opencontainers.image.url": "https://github.com/Mintplex-Labs/anything-llm",
                "org.opencontainers.image.version": "1.2.4"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "ff7262e3e15684ade8b1320cb694646f27f263350e6cfa79b469343432d8e824",
            "SandboxKey": "/var/run/docker/netns/ff7262e3e156",
            "Ports": {
                "3001/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "3001"
                    },
                    {
                        "HostIp": "::",
                        "HostPort": "3001"
                    }
                ]
            },
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "d390509ff076854982122966911939032d7fb7d984dcf2bdf9a5d6c84a666009",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.2",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:02",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "MacAddress": "02:42:ac:11:00:02",
                    "NetworkID": "fb5f18be71e75430b0a2f6fe33230235b7b186994d169f319819f78396deb633",
                    "EndpointID": "d390509ff076854982122966911939032d7fb7d984dcf2bdf9a5d6c84a666009",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "DriverOpts": null,
                    "DNSNames": null
                }
            }
        }
    }
]
[ec2-user@ip-172-31-25-138 ~]$ 
```

**Comment by timothycarambat** [comment]
Oh, now i see. @daaddd 

> Administrators do not have a chat limit since they are trusted

Thats it.
Screenshot 2024-11-19 at 2 00 52 PM
Screenshot 2024-11-19 at 2 01 00 PM
Screenshot 2024-11-19 at 2 02 14 PM

**Comment by daaddd** [comment]
Ah thanks explains it! Thank you!

**Pull request merged**